### PR TITLE
'Be this camera' fixes and improvements

### LINKED ISSUE #10427: [Editor] Using a camera other than editor camera can lead to "Internal logic error"
**Describe the bug**
When using a game camera in the Editor, sometimes the message "[Error] (EditorViewportWidget) - Internal logic error - active view changed to an entity which is not an editor camera. Please report this as a bug." will get printed.

**Steps to reproduce**
1. Open the Editor.
2. Create a new level.
3. Go to "Atom Default Environment" -> "Camera" entity, and choose "Be this camera"
4. Go to the viewport camera widget and change the speed of the camera.
5. Go to File->Save and save the level.
6. Select other entities.
7. Move the camera around a bit.
8. Go back to the viewport camera widget and change the speed of the camera again.
9. Repeat steps 6-8 as necessary.

**Expected behavior**
No errors occur.

**Actual behavior**
The error "[Error] (EditorViewportWidget) - Internal logic error - active view changed to an entity which is not an editor camera. Please report this as a bug." sometimes prints.

**Screenshots/Video**
https://user-images.githubusercontent.com/82224783/176475298-f9dc6f6c-f812-47b4-a151-8e4da10bbd40.mp4

**Found in Branch**
development

### LINKED ISSUE #10428: [Editor] Intermittent error - "[Error] (LocalViewBookmarkLoader) - Couldn't find or create LocalViewBookmarkComponent."
**Describe the bug**
Sometimes when loading levels, the error "[Error] (LocalViewBookmarkLoader) - Couldn't find or create LocalViewBookmarkComponent." occurs.

**Steps to reproduce**
1. Open the Editor.
2. Create a new level.
3. Go to the "Atom Default Environment" -> "Camera" component and choose "Be this camera"
4. Move the camera around a little
5. Change the camera speed in the viewport camera widget.
6. Create another new level, saving when prompted.
7. Follow steps 3-5 again.
8. Open the previous level, saving when prompted.
9. Repeat steps 3-5 and 8 as needed until the error occurs.

**Expected behavior**
No error occurs.

**Actual behavior**
The error "[Error] (LocalViewBookmarkLoader) - Couldn't find or create LocalViewBookmarkComponent." occurs.

**Screenshots/Video**
https://user-images.githubusercontent.com/82224783/176479599-fa54d862-1633-46c0-b998-697687aa9f47.mp4

**Found in Branch**
development

### LINKED ISSUE #13476: Incorrect transforms for Camera component options
**Describe the bug**
The "Match Viewport" and "Be this Camera" options for the Camera component options do not affect the Camera position in the Game Mode. The Camera's view can be modified by the transform options of the Camera entity.

**Steps to reproduce**
1. Create a new default level.
2. Enter the Game Mode.
3. Observe the Camera view.
4. Exit the Game Mode.
5. Select the Camera entity.
6. Select "Match Viewport" or "Be this Camera" options.
7. Change the position of the Camera entity using the options above.
8. Enter the Game Mode again.

**Expected behavior**
The Camera entity shows the updated position.

**Actual behavior**
The Camera entity shows the starting position.

**Screenshots/Video**

https://user-images.githubusercontent.com/86952591/205062876-d9dabdaf-e0a4-444f-b70c-4e53cde16947.mp4

**Found in Branch**
Development.

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
bbf6981

**Desktop/Device:**
Device: PC
OS: Windows
Version 10
CPU AMD Ryzen 5 3600
GPU Nvidia GeForce RTX 2060 SUPER
Memory 16GB

### LINKED ISSUE #13525: Bug Report: Camera editor selection and other cameras move to the wrong position
**Describe the bug**
switching to other cameras the wrong position

**Screenshots/Video**

https://user-images.githubusercontent.com/17591539/205438132-14bc2c61-f242-4302-8aa1-ef893794ccb3.mp4

https://user-images.githubusercontent.com/17591539/205438135-d2cb7ea7-a0d3-4b70-a5f7-53c806a10b01.mp4

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
4ac8adffbe54068a43b7e90d3248b7cf309ee843

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Fedora
 - Version 36

### LINKED ISSUE #13744: (EditorViewportWidget) - Internal logic error
**Describe the bug**
I opened a level, and got this error in the console:
[Error] (EditorViewportWidget) - Internal logic error - active view changed to an entity which is not an editor camera. Please report this as a bug.

**Assets required**
I was using the ROSConDemo project, but I suspect this could be replicated by creating two levels which each have an entity with an entity camera.

**Steps to reproduce**
1. Open the 'Main' level
2. In the entity outliner, select a camera entity
3. In the entity inspecter, click 'Be this camera'
4. Open another level, which also has a entity with a camera
5. See error message in console
6. Go back to the first level
7. Click 'Be this camera' on the entity again
8. The camera remains in the position of the editor camera, not the position from the entity camera

**Expected behavior**
No console error message, and clicking 'Be this camera' on an entity always takes the viewport camera to the position of that camera

**Actual behavior**
Console error message, and camera remains in the position of the editor camera

**Screenshots/Video**

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
37b99bfa3e5e88a1d545c1fe40784531d70766e2

**Desktop/Device (please complete the following information):**
 - Device: PC
 - OS: Windows
 - Version: 10
 - CPU: Intel I9-10900X
 - GPU: NVidia GeForce RTX 2080 SUPER
 - Memory: 64GB

**Additional context**

### LINKED ISSUE #13763: "Be This Camera" Mode Blocks Editing Transform
**Describe the bug**
While in "be this camera mode" I cannot make micro-adjustments to the camera position by changing the values by hand in the transform component.

**Steps to reproduce**
1. Open the default level
3. Select the camera entity
4. Select "Be this camera"
5. Try to edit the camera position by editing the transform component

**Expected behavior**
Editing the transform component should move the camera

**Actual behavior**
User must click "Return to default editor camera" in order to hand-edit the transform 

**Screenshots/Video**
image

Notice in the video I change the X rotation value to 0.0, 2.0, and 10.0 but it always snaps back to the original value.
https://user-images.githubusercontent.com/32776221/208368292-00d626a4-75fc-403a-8016-4a1d2e3b034d.mp4

**Found in Branch**
development

**Commit ID from [o3de/o3de](https://github.com/o3de/o3de) Repository**
47eea0c820a3639fcbcb04aff991235e9132adb1 
Monday December 12 2022

### LINKED ISSUE #14285: [Error] (EditorViewportWidget) - Internal logic error - active view changed to an entity which is not an editor camera. Please report this as a bug.Bug Report
[Error] (EditorViewportWidget) - Internal logic error - active view changed to an entity which is not an editor camera. Please report this as a bug.

### PR DESCRIPTION
**Pull request by hultonha**
## What does this PR do?

Fixes #13525
Fixes #14285
Fixes #13763
Fixes #13744
Fixes #13476
Fixes #10428
Fixes #10427

_These are just the ones I could find... there could be more..._

This PR fixes a _ton_ of issues related to 'Be this camera'... it's been hard to keep track, this really was a rabbit whole.

I'm pretty confident things are much better now than they were before and ~~I'm in the progress of adding tests to verify the corrected functionality~~ (__5 total test have been added__).

The problems included but were not limited to...

1. Going to game mode and back when being Editor camera would not return to the previous position (instead the camera would go to the saved view bookmark 'last known location', and was only updated when saved).
2. Changes to `EditorCameraComponent` would not persist when going to `GameMode` after using 'Be this camera'.
3. View bookmark warning would fire because `SetDefaultCamera` in `EditorViewportWidget` would be called before view bookmarks are available.
4. An error message was printed when changing properties on the camera without it selected when it Edit Mode (`[Error] (EditorViewportWidget) - GetCurrentViewEntityId called while the current view is being changed. You may get inconsistent results if you make use of the returned entity ID. This is an internal error, please report it as a bug.` and `[Error] (EditorViewportWidget) - Internal logic error - active view changed to an entity which is not an editor camera. Please report this as a bug.`).
5. Changing transform component when in 'Be this camera' would not change the camera transform (instead it would stutter).
6. Going to game mode with multiple cameras in the scene would produce `[Warning] (EditorViewportWidget) - Tried to change the editor camera during play game in editor; this is currently unsupported`.
7. Opening a level and just switching between EditorCamera and Camera (CameraComponent) gets the Editor into a bad state.
8. Going to game mode when in 'Be this camera' would print `[Warning] (EditorViewportWidget) - Tried to change the editor camera during play game in editor; this is currently unsupported`.
9. Opening a new level would go to the origin when in normal EditorCamera.
10. Always would return to EditorCamera when starting Game Mode as 'Be in this camera' and then returning to Edit Mode.
11. When in 'Be this camera' and then changing level, the camera would get stuck in a weird state (still assigned to a camera from the previous scene when in EditorCamera mode).

## How was this PR tested?

Extensive manual testing in the Editor (changing cameras, changing level, going in and out of game mode, modifying camera properties, etc...)

New automation tests to verify a significant proportion of the error lists above.

To be certain the tests were catching the correct behavior, I worked in two branches (development and my branch) and made sure the tests failed on development first. I then moved them to my branch to prove the test would then pass with my fixes (see output below).

### Before

```bash
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_BasicEditorWorkflows_LevelEntityComponentCRUD[windows-AutomatedTesting-windows_editor] PASSED       [  4%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_BasicEditorWorkflows_GPU_LevelEntityComponentCRUD[windows-AutomatedTesting-windows_editor] PASSED   [  9%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_Docking_BasicDockedTools[windows-AutomatedTesting-windows_editor] PASSED                              [ 14%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_EntityOutliner_EntityOrdering[windows-AutomatedTesting-windows_editor] PASSED                         [ 19%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraGameModeTransitionWithMultipleCamerasReportsNoErrors[windows-AutomatedTesting-windows_editor] FAILED [ 23%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraBeThisCameraIsClearedWhenChangingLevel[windows-AutomatedTesting-windows_editor] FAILED [ 28%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::run_batched_tests[windows-AutomatedTesting-windows_editor] PASSED                                                      [ 33%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_ParentEntityTransform_Affects_ChildEntityTransform[windows-AutomatedTesting-windows_editor] PASSED [ 38%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_ChildEntityTransform_Persists_After_ParentEntityTransform[windows-AutomatedTesting-windows_editor] PASSED [ 42%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraMovesToEntityTransformWithBeThisCamera[windows-AutomatedTesting-windows_editor] PASSED [ 47%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraReturnsToOriginalPositionWhenTogglingBeThisCamera[windows-AutomatedTesting-windows_editor] PASSED [ 52%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraReturnsToOriginalPositionWhenLeavingGameMode[windows-AutomatedTesting-windows_editor] FAILED [ 57%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraTransformCanBeModifiedWhileInBeThisCamera[windows-AutomatedTesting-windows_editor] FAILED [ 61%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::run_batched_tests[windows-AutomatedTesting-windows_editor] PASSED                                        [ 66%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_AssetPicker_UI_UX[windows-AutomatedTesting-windows_editor] PASSED                                   [ 71%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_BasicEditorWorkflows_ExistingLevel_EntityComponentCRUD[windows-AutomatedTesting-windows_editor] PASSED [ 76%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::run_batched_tests[windows-AutomatedTesting-windows_editor] PASSED                                          [ 80%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_AssetBrowser_SearchFiltering[windows-AutomatedTesting-windows_editor] PASSED                          [ 85%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_AssetBrowser_TreeNavigation[windows-AutomatedTesting-windows_editor] PASSED                           [ 90%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_Menus_EditMenuOptions_Work[windows-AutomatedTesting-windows_editor] PASSED                            [ 95%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_Menus_ViewMenuOptions_Work[windows-AutomatedTesting-windows_editor] PASSED                            [100%]
=============================================================================== 4 failed, 17 passed in 360.17s (0:06:00) ===============================================================================
```

### After

```bash
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_BasicEditorWorkflows_LevelEntityComponentCRUD[windows-AutomatedTesting-windows_editor] PASSED       [  4%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_BasicEditorWorkflows_GPU_LevelEntityComponentCRUD[windows-AutomatedTesting-windows_editor] PASSED   [  9%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_Docking_BasicDockedTools[windows-AutomatedTesting-windows_editor] PASSED                              [ 14%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_EntityOutliner_EntityOrdering[windows-AutomatedTesting-windows_editor] PASSED                         [ 19%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraMovesToEntityTransformWithBeThisCamera[windows-AutomatedTesting-windows_editor] PASSED [ 23%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraReturnsToOriginalPositionWhenTogglingBeThisCamera[windows-AutomatedTesting-windows_editor] PASSED [ 28%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraReturnsToOriginalPositionWhenLeavingGameMode[windows-AutomatedTesting-windows_editor] PASSED [ 33%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraTransformCanBeModifiedWhileInBeThisCamera[windows-AutomatedTesting-windows_editor] PASSED [ 38%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraGameModeTransitionWithMultipleCamerasReportsNoErrors[windows-AutomatedTesting-windows_editor] PASSED [ 42%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_EditorCameraBeThisCameraIsClearedWhenChangingLevel[windows-AutomatedTesting-windows_editor] PASSED [ 47%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::run_batched_tests[windows-AutomatedTesting-windows_editor] PASSED                                        [ 52%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_AssetPicker_UI_UX[windows-AutomatedTesting-windows_editor] PASSED                                   [ 57%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationNoAutoTestMode::test_BasicEditorWorkflows_ExistingLevel_EntityComponentCRUD[windows-AutomatedTesting-windows_editor] PASSED [ 61%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::run_batched_tests[windows-AutomatedTesting-windows_editor] PASSED                                                      [ 66%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_ParentEntityTransform_Affects_ChildEntityTransform[windows-AutomatedTesting-windows_editor] PASSED [ 71%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomation::test_EditorWorkflow_ChildEntityTransform_Persists_After_ParentEntityTransform[windows-AutomatedTesting-windows_editor] PASSED [ 76%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::run_batched_tests[windows-AutomatedTesting-windows_editor] PASSED                                          [ 80%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_AssetBrowser_SearchFiltering[windows-AutomatedTesting-windows_editor] PASSED                          [ 85%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_AssetBrowser_TreeNavigation[windows-AutomatedTesting-windows_editor] PASSED                           [ 90%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_Menus_EditMenuOptions_Work[windows-AutomatedTesting-windows_editor] PASSED                            [ 95%]
AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py::TestAutomationAutoTestMode::test_Menus_ViewMenuOptions_Work[windows-AutomatedTesting-windows_editor] PASSED                            [100%]

==================================================================================== 21 passed in 491.46s (0:08:11) ====================================================================================
```

### CODE REVIEW

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp [hunk 1]
```diff
@@ -207,12 +226,22 @@ namespace Camera
...
-                    &EditorCameraRequests::SetViewAndMovementLockFromEntityPerspective, entityId, lockCameraMovement);
+
+                const AZ::EntityId entityId = selectionModel()->currentIndex().data(Qt::CameraIdRole).value<AZ::EntityId>();
+                // only check entity validity if entity is valid, otherwise the change event will be for the editor camera
+                if (entityId.IsValid())
+                {
+                    // if the entity is not in an active state we are most likely in a transition event (going to game mode
+                    // or changing level) and we do not want so signal any changes to the camera request bus
+                    const AZ::Entity* entity = AzToolsFramework::GetEntityById(entityId);
+                    if (!entity || (entity && entity->GetState() != AZ::Entity::State::Active))
```
Comment (moraaar) [inline Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:238]: nit: The second `entity` check is redundant, the second part of the || would only be done if entity is valid.

````
if (!entity || entity->GetState() != AZ::Entity::State::Active)
````

File: AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py [hunk 2]
```diff
@@ -93,8 +93,38 @@ class TestAutomation(EditorTestSuite):
...
+        file_system.delete([os.path.join(workspace.paths.engine_root(), "AutomatedTesting", "Levels", "tmp_level_2")],
+                        True, True)
+
     class test_EditorWorkflow_ParentEntityTransform_Affects_ChildEntityTransform(EditorBatchedTest):
         from .EditorScripts import EditorWorkflow_ParentEntityTransform_Affects_ChildEntityTransform as test_module
 
     class test_EditorWorkflow_ChildEntityTransform_Persists_After_ParentEntityTransform(EditorBatchedTest):
         from .EditorScripts import EditorWorkflow_ChildEntityTransform_Persists_After_ParentEntityTransform as test_module
+
+    class test_EditorWorkflow_EditorCameraMovesToEntityTransformWithBeThisCamera(EditorSingleTest):
```
Comment (moraaar) [inline AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py:108]: There are 6 new test files but only 5 added here. Missed to add the test `EditorWorkflow_EditorCameraGameModeTransitionWithMultipleCamerasReportsNoErrors` to this file?

**Review by moraaar — APPROVED** [review]
LGTM. Great work @hultonha! It's great that the camera behavior is going to be much stable after this change.

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/EditorWorkflow_EditorCameraBeThisCameraIsClearedWhenChangingLevel.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        "Initial camera positions match in first and second level",
+        "Initial camera positions do not match in first and second level",
+    )
+    entity_position_changed_when_not_in_be_this_camera = (
+        "Entity position did not change when setting view position while not in 'Be this camera'",
+        "Entity position did change when setting view position while not in 'Be this camera'"
+    )
+    view_position_changed_when_moving_entity_when_not_in_be_this_camera = (
+        "View position did not change when moving entity while not in 'Be this camera' mode",
+        "View position did change when moving entity while not in 'Be this camera' mode",
```
Comment (mbalfour-amzn) [inline AutomatedTesting/Gem/PythonTests/editor/EditorScripts/EditorWorkflow_EditorCameraBeThisCameraIsClearedWhenChangingLevel.py:28]: Suggestion: These could be written in a way where the failure case is more obviously a failure, possibly even using the word "failed". For example:
"Initial camera positions failed to match in first and second level"
"Entity position failed to remain the same when setting view position..."
"View position failed to remain the same when moving entity..."

It would also be nice if the success case avoided using negatives to make it even more obvious which statement represents success and which represents failure. For example:
"Entity position remained the same when setting view position outside of 'Be this camera' mode"
"View position remained then same when moving entity outside of 'Be this camera' mode"

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp [hunk 4]
```diff
@@ -98,22 +97,32 @@ namespace Camera
 
         void CameraListModel::OnCameraAdded(const AZ::EntityId& cameraId)
         {
-            // If the camera entity is not an editor camera entity, don't add it to the list.
-            // This occurs when we're in simulation mode.
-
-            //We reset the m_firstEntry value so we can update m_lastActiveCamera when we remove from the cameras list
-            m_firstEntry = true;
+            // if the camera entity is not an editor camera entity, don't add it to the list
+            // this occurs when we're in simulation mode.
```
Comment (mbalfour-amzn) [inline Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:101]: Nit: The capitalization / punctuation was actually correct before the change. Not sure why the "if" and "this" became decapitalized and the period after "list" was removed.

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp [hunk 5]
```diff
@@ -161,8 +173,13 @@ namespace Camera
             return index(row, 0);
         }
 
+        void CameraListModel::Reset()
+        {
+            m_lastActiveCamera.SetInvalid();
+            m_cameraItems = AZStd::vector<CameraListItem>(1, CameraListItem(AZ::EntityId()));
```
Comment (mbalfour-amzn) [inline Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:179]: Suggestion: might be nice to add a comment explaining why m_cameraItems defaults to a list of 1 item with an invalid EntityId instead of an empty list.

**Review by mbalfour-amzn — APPROVED** [review]

**Comment by Kadino** [comment]
> I have duplicated a useful function across a few of my tests...
> 
> and I'd ideally like to move this to a util, but I'm not sure where would be the best place for it to live....
> 
> I'm thinking a static function on `EditorTestHelper` might be a good fit? Please let me know if you have any other suggestions. Having this util return a `math.Vector3` is super useful to be then able to call `.IsClose(...)` for example immediately. Thanks!

I'm mildly surprised that the vector needs to be repacked. Doesn't [get_current_view_position already return a Vector3](https://github.com/o3de/o3de/blob/60adc098da2d3026280bd0caa10915d6c8e6e14f/Code/Editor/CryEditPy.cpp)? Is something preventing directly calling IsClose?

**Comment by hultonha** [comment]
> > I have duplicated a useful function across a few of my tests...
> > and I'd ideally like to move this to a util, but I'm not sure where would be the best place for it to live....
> > I'm thinking a static function on `EditorTestHelper` might be a good fit? Please let me know if you have any other suggestions. Having this util return a `math.Vector3` is super useful to be then able to call `.IsClose(...)` for example immediately. Thanks!
> 
> I'm mildly surprised that the vector needs to be repacked. Doesn't [get_current_view_position already return a Vector3](https://github.com/o3de/o3de/blob/60adc098da2d3026280bd0caa10915d6c8e6e14f/Code/Editor/CryEditPy.cpp)? Is something preventing directly calling IsClose?

It does not return a `Vector3`, it seems to be some kind of property bag (sorry I don't know the exact Python term, I don't think it's strictly a Tuple either). This is what massively tripped be up before when trying to use it. So we do need to add this function, I'd just like to know where. Thanks!

**Comment by amzn-rhhong** [comment]
LGTM. Awesome work @hultonha

**Review by amzn-rhhong — APPROVED** [review]

**Comment by Kadino** [comment]
> It does not return a `Vector3`

Interesting! I'm not quite familiar enough with why the `AZ::Vector3` from [CryEditPy.cpp](https://github.com/o3de/o3de/blob/60adc098da2d3026280bd0caa10915d6c8e6e14f/Code/Editor/CryEditPy.cpp#L212) gets reflected as an object with some attributes, instead of [reflecting a Vector3 ](https://github.com/o3de/o3de/blob/76e09a36d900e871d170ce1ff6530c85e047093c/Code/Framework/AzCore/AzCore/Math/Vector3.cpp#L293)as constructed in Python with `azlmbr.math.Vector3()` ...but I think that difference is the core issue. Mitigating that by using a repacking helper everywhere isn't a great solution.

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp (hunk 1 above)
Comment (hultonha) [reply Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:238]: Ah good spot! Will update :+1:

File: AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py (hunk 2 above)
Comment (hultonha) [reply AutomatedTesting/Gem/PythonTests/editor/TestSuite_Main.py:108]: Oh good spot! 😬 I'll add that now, good eye thanks!

File: AutomatedTesting/Gem/PythonTests/editor/EditorScripts/EditorWorkflow_EditorCameraBeThisCameraIsClearedWhenChangingLevel.py (hunk 3 above)
Comment (hultonha) [reply AutomatedTesting/Gem/PythonTests/editor/EditorScripts/EditorWorkflow_EditorCameraBeThisCameraIsClearedWhenChangingLevel.py:28]: Hmm I mean you will get a big red failure output anyway right? I think it's obvious that something has gone wrong when the test fails (the condition isn't met), adding another 'failed' to the message isn't going to make a great deal of difference in my mind.

I'll have a look if I can simplify the wording slightly but again I think the test failing and the context of both messages when debugging the test is sufficient.

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp (hunk 4 above)
Comment (hultonha) [reply Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:101]: I routinely do not bother with capitalization and periods in .cpp files and reserve that style for API comments - this style is prevalent in quite a few files (to indicate guide comments) and as I'm now the de-facto maintainer of this file I updated the style while making the other edits/changes - I'd prefer to leave as-is if you don't mind

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp (hunk 5 above)
Comment (hultonha) [reply Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:179]: This predates me and is to handle the normal 'Editor Camera' (not tied to an Entity/Component) in the level - I'll add a comment to explain 👍

**Comment by hultonha** [comment]
> > It does not return a `Vector3`
> 
> Interesting! I'm not quite familiar enough with why the `AZ::Vector3` from [CryEditPy.cpp](https://github.com/o3de/o3de/blob/60adc098da2d3026280bd0caa10915d6c8e6e14f/Code/Editor/CryEditPy.cpp#L212) gets reflected as an object with some attributes, instead of [reflecting a Vector3 ](https://github.com/o3de/o3de/blob/76e09a36d900e871d170ce1ff6530c85e047093c/Code/Framework/AzCore/AzCore/Math/Vector3.cpp#L293)as constructed in Python with `azlmbr.math.Vector3()` ...but I think that difference is the core issue. Mitigating that by using a repacking helper everywhere isn't a great solution.

Okay in that case I'd really leave that larger refactor/change to another PR - I really don't want to start making wider changes at this point so I'll keep the util functions I added local to the file and make a GitHub Issue (GHI) to follow-up and see if we can improve the legacy call (this is an ancient hold-over from Cry unfortunately). I'll update this comment with a link to the GHI once I've created it.

File: Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp (hunk 1 above)
Comment (hultonha) [reply Gems/Camera/Code/Source/ViewportCameraSelectorWindow.cpp:238]: Test re-added and description updated with output from missed test, thanks!

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp [hunk 6]
```diff
@@ -276,19 +269,26 @@ namespace AzToolsFramework
...
             {
                 return Write(genericStream, stringBuffer);
             });
 
         AZ_Warning(
-            "LocalViewBookmarkLoader", saved, R"(Unable to save Local View Bookmark file to path "%.*s"\n)", AZ_STRING_ARG(localBookmarksFileName.Native()));
+            "LocalViewBookmarkLoader",
+            saved,
+            R"(Unable to save Local View Bookmark file to path "%.*s"\n)",
+            AZ_STRING_ARG(localBookmarksFileName.Native()));
```
Comment (lawsonamzn) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:291]: AZ_Warning is defined as nothing when AZ_ENABLE_TRACING is not set.  Its unset in RELEASE.  Which is maybe fine for Tools Framework code, if we ever want to ship it in release, though, this would be an issue.

This will cause this not to actually try to save if its ever hit in release.   Is this ok?  I assume this is why you added [[maybe_unused]] so it might be okay depending on what/when you expect this code to run.  It looks like its actually doing work though.

**Review by lawsonamzn — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp (hunk 6 above)
Comment (hultonha) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp:291]: I'll look into this and double check. I think this is likely a problem we'd hit in lots of code in AzToolsFramework as I don't think we ever compile it in Release and so don't hit these unused variable warnings.
